src/syscheckd/create_db.c

```diff
@@ -598,6 +624,10 @@ static int read_file(const char *file_name, int dir_position, whodata_evt *evt,
             free(alert_msg);
             free(c_sum);
         }
+        if (statbuf_lnk) {
```

Comment:
This code produces a memory leak at line 587.
Is it necessary to use dynamic memory?